client/lib/reader-feed-subscriptions/index.js

```diff
@@ -1,381 +1,170 @@
 // Reader Feed Subscription Store
 
-//const debug = require( 'debug' )( 'calypso:reader:feed-subs' );
-
-// External Dependencies
-const Dispatcher = require( 'dispatcher' ),
-	reject = require( 'lodash/collection/reject' ),
-	findLast = require( 'lodash/collection/findLast' ),
-	get = require( 'lodash/object/get' ),
-	last = require( 'lodash/array/last' ),
-	Immutable = require( 'immutable' ),
-	clone = require( 'lodash/lang/clone' ),
-	map = require( 'lodash/collection/map' );
-
-// Internal Dependencies
-const Emitter = require( 'lib/mixins/emitter' ),
-	ActionTypes = require( './constants' ).action,
-	ErrorTypes = require( './constants' ).error,
-	PostStoreActionTypes = require( 'lib/feed-post-store/constants' ).action,
-	FeedSubscriptionHelper = require( './helper' ),
-	States = require( './constants' ).state;
-
-const subscriptionsTemplate = {
-	list: Immutable.List(), // eslint-disable-line new-cap
-	count: 0
+// External dependencies
+import { Map, fromJS } from 'immutable';
+import debugModule from 'debug';
+
+// Internal dependencies
+import { action as actionTypes, state as stateTypes } from './constants';
+import { createReducerStore } from 'lib/store';
+import FeedSubscriptionHelper from './helper';
+
+const debug = debugModule( 'calypso:reader-feed-subs' ); //eslint-disable-line no-unused-vars
+
+const initialState = fromJS( {
+	subscriptions: [],
+	errors: [],
+	perPage: 50,
+	currentPage: 0,
+	isLastPage: false,
+	isFetching: false,
+	subscriptionCount: 0,
+} );
+
+const subscriptionTemplate = Map( { // eslint-disable-line new-cap
+	state: stateTypes.SUBSCRIBED
+} );
+
+const FeedSubscriptionStore = createReducerStore( ( state, payload ) => {
+	switch ( payload.action.type ) {
+		case actionTypes.FOLLOW_READER_FEED:
+			return addSubscription( state, payload.action.data );
+
+		case actionTypes.UNFOLLOW_READER_FEED:
+			return removeSubscription( state, payload.action.data );
+	}
+
+	return state;
+}, initialState );
+
+FeedSubscriptionStore.isFetching = function() {
+	const state = FeedSubscriptionStore.get();
+	return state.get( 'isFetching' );
+};
+
+FeedSubscriptionStore.getLastError = function() {
+	const state = FeedSubscriptionStore.get();
+	return state.has( 'errors' ) ? state.get( 'errors' ).last() : null;
+};
+
+FeedSubscriptionStore.isLastPage = function() {
```

Comment:
if we go es6, these could be `() => FeedSubscriptionStore.get().get( 'isLastPage' )`